src/backend/optimizer/plan/orca.c

```diff
@@ -241,7 +241,6 @@ optimize_query(Query *parse, ParamListInfo boundParams)
 	result->relationOids = glob->relationOids;
 	result->invalItems = glob->invalItems;
 	result->oneoffPlan = glob->oneoffPlan;
-	result->transientPlan = glob->transientPlan;
```

Comment:
1. Why have you removed that? What the `glob->transientPlan;` stands for?
2. Why have you taken "transient" word for namings?  How it represents the need for cache invalidation, how it represents the focuses of the patch?

Reply:
1. in the case of ORCA, glob->transientPlan was always false. now this flag is set directly in the PlannedStmt. https://github.com/arenadata/gpdb/blob/f4c4ee25754a85b824ce23087dc873b7c907e54d/src/backend/optimizer/plan/orca.c#L101

Reply:
2. I have no idea why this word is used, but it is used in the postgres planner. this flag is also used to invalidate the cache of plans created using PREPARE.
https://github.com/arenadata/gpdb/blob/f4c4ee25754a85b824ce23087dc873b7c907e54d/src/backend/utils/cache/plancache.c#L1034-L1038
https://github.com/arenadata/gpdb/blob/f4c4ee25754a85b824ce23087dc873b7c907e54d/src/backend/utils/cache/plancache.c#L832-L839
in the case of ORCA, we also need to invalidation its own cache because if we don't add an index to the relation, it will be cached without it and we won't know about this index until the cache is drop (in case of catalog changes). the postgres planner does not have its own cache.